docs/HowSwiftImportsCModules.md

```diff
@@ -0,0 +1,1005 @@
+properties. Bitfields are mapped to computed Swift properties. Swift structs
+also get a synthesized default initializer (that sets all properties to zero),
+and an elementwise initializer (that sets all properties to the provided
+values).
+
+```c
+// C header.
+
+struct Point {
+  int x;
+  int y;
+};
+
+struct Line {
+  struct Point start;
+  struct Point end;
+  unsigned int brush : 4;
+  unsigned int stroke : 3;
+};
+```
+
+```swift
+// C header imported in Swift.
+
+struct Point {
+  var x: CInt { get set }
+  var y: CInt { get set }
+  init()
+  init(x: CInt, y: CInt)
+}
+
+struct Line {
+  var start: Point { get set }
+  var end: Point { get set }
+  var brush: CUnsignedInt { get set }
+  var stroke: CUnsignedInt { get set }
+
+  // Default initializer that sets all properties to zero.
+  init()
+
+  // Elementwise initializer.
+  init(start: Point, end: Point, brush: CUnsignedInt, stroke: CUnsignedInt)
+}
+```
+
+Swift can also import unnamed and anonymous structs.
+
+```c
+// C header.
+
+struct StructWithAnonymousStructs {
+  struct {
+    int x;
+  };
+  struct {
+    int y;
+  } containerForY;
+};
+```
+
+```swift
+// C header imported in Swift.
+struct StructWithAnonymousStructs {
+  struct __Unnamed_struct___Anonymous_field0 {
+    var x: Int32
+    init()
+    init(x: Int32)
+  }
+  struct __Unnamed_struct_containerForY {
+    var y: Int32
+    init()
+    init(y: Int32)
+  }
+  var __Anonymous_field0: StructWithAnonymousStructs.__Unnamed_struct___Anonymous_field0
+  var x: Int32
+  var containerForY: StructWithAnonymousStructs.__Unnamed_struct_containerForY
+
+  // Default initializer that sets all properties to zero.
+  init()
+
+  // Elementwise initializer.
+  init(
+    _ __Anonymous_field0: StructWithAnonymousStructs.__Unnamed_struct___Anonymous_field0,
+    containerForY: StructWithAnonymousStructs.__Unnamed_struct_containerForY
+  )
+}
+```
+
+See also Apple's documentation about this topic: [Using Imported C Structs and
+Unions in
+Swift](https://developer.apple.com/documentation/swift/imported_c_and_objective-c_apis/using_imported_c_structs_and_unions_in_swift).
+
+# Unions
+
+Swift does not have a direct equivalent to a C union. C unions are mapped to
+Swift structs with computed properties that read from/write to the same
+underlying storage.
+
+```c
+// C header.
+
+union IntOrFloat {
+  int i;
+  float f;
+};
+```
+
+```swift
+// C header imported in Swift.
+
+struct IntOrFloat {
+  var i: CInt { get set } // Computed property.
+  var f: Float { get set } // Computed property.
+  init(i: CInt)
+  init(f: Float)
+  init()
+}
+```
+
+See also Apple's documentation about this topic: [Using Imported C Structs and
+Unions in
+Swift](https://developer.apple.com/documentation/swift/imported_c_and_objective-c_apis/using_imported_c_structs_and_unions_in_swift).
+
+# Enums
+
+C enums are mapped to Swift structs or enums.
+
+C enums are quirky at least in two ways:
```

Comment:
I feel like this is burying the important part of enums---that C enums are mapped into Swift enums---way at the end. Perhaps switch the order here, so you describe the basic mapping and *then* go into frozen vs. non-frozen?

Reply:
The issue I'm trying to get at is that C enums (plain ones, not `NS_ENUM`s) are mapped to Swift structs, and I feel like I need to explain why.

I added an example of an mapping C enums to Swift structs upfront in this section, to motivate the reader to read through the long discussion.